### PR TITLE
Run integration test from pr

### PR DESCRIPTION
**Pull request by MartinBasti**
## Maintainers will complete the following section

- [x] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [x] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/konflux-ci/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

File: gitops/snapshot.go [hunk 1]
```diff
@@ -696,6 +702,11 @@ func PrepareSnapshot(ctx context.Context, adapterClient client.Client, applicati
 	}
 	snapshot := NewSnapshot(application, &snapshotComponents)
 
+	// expose the source repo URL in the snapshot as annotation do we don't have to do lookup in integration tests
+	if newComponentSource.GitSource != nil {
+		metadata.SetAnnotation(snapshot, SnapshotGitSourceRepoURLAnnotation, newComponentSource.GitSource.URL)
```
Comment (dirgim) [inline gitops/snapshot.go:707]: The [gosec check](https://github.com/konflux-ci/integration-service/actions/runs/9259151522/job/25470527489?pr=747) is complaining about this line not handling the error that's returned by `metadata.SetAnnotation`.

File: internal/controller/snapshot/snapshot_adapter.go [hunk 2]
```diff
@@ -447,21 +447,77 @@ func (a *Adapter) createMissingReleasesForReleasePlans(application *applicationa
...
+	if urlVal, ok := params[tekton.TektonResolverGitParamURL]; ok {
+		// urlVal may or may not have git suffix specified :')
+		return urlToGitUrl(urlVal) == urlToGitUrl(annotations[gitops.PipelineAsCodeRepoURLAnnotation])
+	}
+
+	// undefined state, no idea what was configured in resolver, don't touch it
+	return false
+}
+
+func getGitResoverUpdateMap(snapshot *applicationapiv1alpha1.Snapshot) map[string]string {
```
Comment (dirgim) [inline internal/controller/snapshot/snapshot_adapter.go:493]: typo: `getGitResoverUpdateMap` -> `getGitResolverUpdateMap`

**Review by dirgim — APPROVED** [review]
LGTM apart from the tiny typo

File: internal/controller/snapshot/snapshot_adapter.go (hunk 2 above)
Comment (MartinBasti) [reply internal/controller/snapshot/snapshot_adapter.go]: fixed

**Review by dirgim — APPROVED** [review]

**Review by jsztuka — APPROVED** [review]
lgtm

**Pull request merged**